### PR TITLE
Prevent Addition of Duplicate Alarms for Same Time #244

### LINKED ISSUE #244: Bug: Prevent Addition of Duplicate Alarms for Same Time
### Issue Description:
When users attempt to set multiple alarms for the same time (e.g., 6:00 AM) within the app, the observed behavior is that only the latest alarm among them triggers, while the other alarms are scheduled for the next day

#### Proposed Solution:
if attempting to set multiple alarms for the same time, the app should display an alert message indicating that an alarm for that specific time is already set, preventing the addition of duplicate alarms at that time

Could you please assign this issue to me?

### Screenshots

WhatsApp Image 2023-12-17 at 2 06 08 PM
WhatsApp Image 2023-12-17 at 2 06 08 PM-2

### PR DESCRIPTION
**Pull request by Deepanshuigtm**
### Description
When users attempt to set multiple alarms for the same time (e.g., 6:00 AM) within the app, the observed behavior is that only the latest alarm among them triggers, while the other alarms are scheduled for the next day

### Proposed Changes
if attempting to set multiple alarms for the same time, the app should display an alert message indicating that an alarm for that specific time is already set, preventing the addition of duplicate alarms at that time

## Fixes #244 

#244 with the issue number which is fixed in this PR
## Screenshots

WhatsApp Image 2023-12-18 at 5 52 36 PM

WhatsApp Image 2023-12-18 at 5 52 42 PM

## Checklist

- [x] Tests have been added or updated to cover the changes
- [x] Documentation has been updated to reflect the changes
- [x] Code follows the established coding style guidelines
- [x] All tests are passing

### CODE REVIEW

File: lib/app/modules/addOrUpdateAlarm/views/add_or_update_alarm_view.dart [hunk 1]
```diff
@@ -122,633 +125,717 @@ class AddOrUpdateAlarmView extends GetView<AddOrUpdateAlarmController> {
...
-                              controller.selectedTime.value,
-                            ),
-                          ),
-                          intervalToAlarm: Utils.getMillisecondsToAlarm(
-                            DateTime.now(),
-                            controller.selectedTime.value,
+                      backgroundColor: themeController.isLightMode.value
+                          ? kLightSecondaryBackgroundColor
+                          : ksecondaryBackgroundColor,
+                      title: 'Alarm already set for $time',
```
Comment (MarkisDev) [inline lib/app/modules/addOrUpdateAlarm/views/add_or_update_alarm_view.dart:185]: I think this dialog might be bad for ux, it should ideally ask the user if they want to overwrite the settings for the new alarm (considering they might want something else enabled).

File: lib/app/modules/addOrUpdateAlarm/views/add_or_update_alarm_view.dart [hunk 2]
```diff
@@ -122,633 +125,717 @@ class AddOrUpdateAlarmView extends GetView<AddOrUpdateAlarmController> {
...
+                  } else {
+                    controller.offsetDetails.value = {};
+                  }
+
+                  String time = Utils.timeOfDayToString(
+                    TimeOfDay.fromDateTime(
+                      controller.selectedTime.value,
+                    ),
+                  );
+                  bool res = await IsarDb.doesAlarmExistTime(time,true);
```
Comment (MarkisDev) [inline lib/app/modules/addOrUpdateAlarm/views/add_or_update_alarm_view.dart:176]: Also, this doesn't account for shared alarms right? I could have an alarm in firestore and another in isar with the same time, that would have to be fixed as well.

**Review by MarkisDev — CHANGES_REQUESTED** [review]
I assume you've missed the edge cases with shared alarms. There could be a shared alarm set for a "main time" which conflicts with another alarm set locally in isar. 

However, always rely on offsetted time since that is what will eventually be the right time for the alarm to ring for each shared user.

File: lib/app/modules/addOrUpdateAlarm/views/add_or_update_alarm_view.dart (hunk 2 above)
Comment (Deepanshuigtm) [reply lib/app/modules/addOrUpdateAlarm/views/add_or_update_alarm_view.dart:176]: ok I will fix it

**Pull request closed without merging**